### PR TITLE
Re-use scanner when reading docker logs for node discovery

### PR DESCRIPTION
**Pull request by iogakos**
Fixes an issue where GetLogLine would return invalid JSON lines

### CODE REVIEW

File: flow/flow.go [hunk 1]
```diff
@@ -428,7 +423,11 @@ func (d *Flow) updateFromLogs(containerName string) error {
 		}
 
 		m := map[string]interface{}{}
-		if err := json.Unmarshal(got, &m); err != nil {
+		// This assumes the container is not using a TTY. In this case
+		// stdout/stderr are multiplexed on the same stream and 8-byte header
+		// precedes each line. We don't need to parse the header since we are
+		// using scanner.Bytes().
+		if err := json.Unmarshal(got[8:], &m); err != nil {
```
Comment (tomasger) [inline flow/flow.go:430]: Let's add a length check to not cause an index out of bounds panic in case of something unexpected.

**Review by tomasger — CHANGES_REQUESTED** [review]

File: flow/flow.go (hunk 1 above)
Comment (iogakos) [reply flow/flow.go:430]: Addressed in 7a9bb63

**Review by tomasger — APPROVED** [review]

File: flow/flow.go [hunk 2]
```diff
@@ -475,6 +462,30 @@ func (d *Flow) updateFromLogs(containerName string) error {
...
 
+// access nodes (and potentially others) use chain_id to log the network
+var networkLogKeyCandidates = []string{"chain", "chain_id"}
+
+// updateNetworkFromJSONLog update the network by looking for a known value
+// in a list of keys. Network is updated based on the first key with a value
+// containing with a known network.
+func (d *Flow) updateNetworkFromJSONLog(m map[string]interface{}) {
+	for _, key := range networkLogKeyCandidates {
+		if chain, ok := m[key]; ok && d.network == "" {
```
Comment (tomasger) [inline flow/flow.go:473]: I'd suggest extracting the conditional `d.network == ""` to the beginning of a function for an early return.

Comment (iogakos) [reply flow/flow.go]: Addressed in 44febfa3a5

File: internal/pkg/discover/utils/utils.go [hunk 3]
```diff
@@ -208,3 +209,33 @@ func getDockerClient() (*client.Client, error) {
...
+	networkBenchnet  = "benchnet"
+)
+
+// KnownNetworks list valid network strings
+var KnownNetworks = []string{
+	string(networkMainnet),
+	string(networkLocalnet),
+	string(networkTestnet),
+	string(networkCanarynet),
+	string(networkBenchnet),
```
[the remark this answers is not in the document]
Comment (iogakos) [reply internal/pkg/discover/utils/utils.go]: Addressed in 44febfa3a5

**Review by tomasger — APPROVED** [review]

**Pull request merged**